src/src/PluginStructs/P122_data_struct.cpp

```diff
@@ -418,7 +428,7 @@ bool P122_data_struct::requestHumidity()
 
 //////////////////////////////////////////////////////////////////////////////////////////////////
 // Read temperature measurement results from device
-bool P122_data_struct::readTemperature()
+bool P122_data_struct::readValue(uint16_t &value)
```

Comment:
You aren't passing the 30 or 90 register identifier that's also a difference between the 2 functions. May be helpful to use a #define or 2 to pass that in, for readability:
```
#define P122_READ_TEMPERATURE 90
#define P122_READ_HUMIDITY    30
```
then use like `if (!readValue(P122_READ_TEMPERATURE, _rawTemperature))` etc.

Reply:
Copied code and just realize it is a timeout value. Looking at teh datasheets it is the complete measurement cycle which differes between humidity and temperature. But thgis should already be handled by the state machine and readBytes() should be called after the conversion. I think it will complicate the code when differentiating between temperature & humidity, so I define one timeout value. Could be much less than 90, see before, but let's be conservative.